### PR TITLE
C scopes collections

### PR DESCRIPTION
**Pull request by abhayayay**
#### Fixes #.

- [ ] Ran `flake8`
- [ ] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- 
-

### CODE REVIEW

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+         @summary: Creating scope and collections with space in names
+        '''
+        db = self.db_obj.create(db_name)
+        try:
+            self.db_obj.createCollection(db, collection, scope)
+            created_Collection = self.db_obj.collectionObject(db, collection, scope)
+            created_CollectionName = self.collection_obj.collectionName(created_Collection)
+            assert created_CollectionName == collection, "Scope and collection created but not found"
+        except:
+            print("Invalid scope and collection name")
```
Comment (gilkal) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:27]: Maybe add the invalid name that's failing? That way it'll be easier to get the relevant string just from reading the logs.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import pytest
```
Comment (gilkal) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:1]: Maybe it's worth adding a setup/teardown fixture so everything that's created is cleanup at the end of test. That way, in the future it'll be easier to remove the clusters reset at the beginning of every test.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        total_collections = []
+        for i in range(1, no_of_scope + 1):
+            scopeName = "scope-" + str(i)
+            total_collections.extend(self.collection_obj.allCollection(db, scopeName))
+        assert len(total_collections) == (collection_per_scope * no_of_scope), " Missing scopes and collections "
+
+    @pytest.mark.parametrize("db_name, scope, collection", [
+        (random_string(6), random_string(6), random_string(6))
+    ])
+    def test_delete_non_existant_scopes_collections(self, db_name, scope, collection):
```
Comment (gilkal) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:78]: Is this tested in the development tests?

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        collections = []
+        for i in range(1, no_of_scope + 1):
+            scopeName = "scope-" + str(i)
+            collections.append(self.db_obj.collectionObject(db, collection, scopeName))
+        assert len(collections) == no_of_scope, "Number of collections made, mismatch"
+
+    @pytest.mark.parametrize("db_name, no_of_scope, no_of_collection", [
+        (random_string(6), random.randrange(1000), random.randrange(1000))
+    ])
+    def test_create_more_than_1000_scopes_collections(self, db_name, no_of_scope, no_of_collection):
```
[the remark this answers is not in the document]
Comment (abhayayay) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:54]: We can add that for to check the accessibility of all the scopes and collections we created.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py (hunk 2 above)
Comment (abhayayay) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:1]: The database is deleted at teardown from conftest.py. We can create a custom fixture for individual tests.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py (hunk 3 above)
Comment (abhayayay) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:78]: not mentioned in test plan.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            try:
+                self.db_obj.createCollection(db, collection, scopeName)
+            except:
+                print("Failed to create collection in scope " + scopeName)
+                assert False
+        collections = []
+        for i in range(1, no_of_scope + 1):
+            scopeName = "scope-" + str(i)
+            collections.append(self.db_obj.collectionObject(db, collection, scopeName))
+        assert len(collections) == no_of_scope, "Number of collections made, mismatch"
```
[the remark this answers is not in the document]
Comment (abhayayay) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_scopesCollections.py:54]: The case checks that if we are able to create same collections at different scopes and if all such collections are accessible or not.

**Review by gilkal — APPROVED** [review]

**Pull request merged**